project/season06/README.md

```diff
@@ -0,0 +1,91 @@
+    FFT is an algorithm that computes the DFT of a sequence, or its inverse (IDFT).  
+
+    Computing DFT directly from the definition is too slow to be practical. FFT rapidly computes such transformations by factorizing the DFT matrix into a product of sparse factors. As a result, it reduces the complexity of computing the DFT from `O(N^2)` to `O(NlogN)`, which `N` is the data size.  
+
+    - Cooley-Tukey FFT Algorithm  
+        Cooley-Tukey Algorithm is the most common FFT algorithm which uses divide and conquer to recursively decompose a DFT of length N = N1N2 into two shorter sequences of length N1 and N2.
+        > <p align="center"><img src="./images/cooley_tukey_fft.svg" alt="cooley_tukey_fft"/></p>
+
+## Prospective Users
+
+- Audio maker  
+    Creators can use Audio Enhancer to improve audio quality online.
+- Users who use Real-time Meetings  
+    If the system can read the audio in real time and quickly judge and filter the noise, it can help audience hear more clearly.
+
+
+## System Architecture
+<img src="./images/system_architecture.png" alt="system_architecture"/>
+
+## API Description
+`equalize(file, mode='allpass', freq=0, gain=0, Q=0)`
+
+- `file`: An audio file, allow type: `.wav`
+- `mode`: Define the shape of the processing curve.
+    - `lowpass`
+    - `highpass`
+    - `lowshelf`
+    - `highshelf`
+    - `bandpass`
+    - `notch`
+    - `allpass`
+- `freq`: A frequency position need to be adjust. (unit: Hz)
+- `gain`: Amplification or attenuation of a certain range of frequencies. (unit: Db)
+- `Q`: Define the steepness or width of each filter.
+
+The following table describe the meaning of the parameters in each mode:
+
+| `mode`      | description       | `freq`                             | `gain` | `Q` |
+| :---------- | :---------------- | :--------------------------------- | :----- | :------- |
+| `lowpass`   | Low-pass filter   | The cut-off frequency              | unused | Drop-off steepness at the cut-off frequency |
+| `highpass`  | High-pass filter  | The cut-off frequency              | unused | Drop-off steepness at the cut-off frequency |
+| `lowshelf`  | Low-shelf filter  | The upper limit of the frequencies | The amplification or attenuation level | unused |
+| `highshelf` | High-shelf filter | The lower limit of the frequencies | The amplification or attenuation level | unused |
+| `bandpass`  | Band-pass filter  | Center of the frequency band       | unused | The width of the frequency band |
+| `notch`     | Notch filter      | Center of the frequency band       | unused | The width of the frequency band |
+| `allpass`   | All-pass filter   | unused | unused | unused |
+
+
+## Engineering Infrastructure
+
+- Automatic build system: `make`
+- Version control: `git`
+- Testing framework: `pytest`
+- Documentation: Update to GitHub README.md
+- Continuous integration: Github Action
+
+
+## Schedule
+
+<img src="./images/gantt.png" alt="gantt"/>
```

Comment:
It's not a bad thing to practice drawing a Gantt's chart but it is rarely useful in development.  A simple list of schedule items works better for communication in a project of this scale.

Developing unit testing after feature is OK, but if you find the feature development stalls don't hesitate to trying to bring the testing earlier.  Sometimes it helps you unlock the feature development.

When you attempt to optimize for performance, it will be good to compare with an existing implementation.